### ISSUE TITLE
`@stylistic/jsx-max-props-per-line`: fix children on new line

### ISSUE THREAD

**Issue by flippidippi** [eslint-stylistic/eslint-stylistic#869]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

I'm not sure if this is really a fix for this rule or it this would be in a different or new rule.
Currently when running fix for `@stylistic/jsx-max-props-per-line`, it's outputting undesirable output.
It's putting the children on the same line as the closing bracket instead of a new line.

With the following settings:
```js
      '@stylistic/indent': ['error', 2],
      '@stylistic/jsx-first-prop-new-line': ['error', 'multiline-multiprop'],
      '@stylistic/jsx-max-props-per-line': ['error', { maximum: { single: 2, multi: 1 }, }],
      '@stylistic/jsx-closing-bracket-location': ['error', 'tag-aligned'],
      '@stylistic/jsx-closing-tag-location': 'error'
```

Running fix on this code:
```tsx
    <div id='' className='' title=''>Child</div>
```

We get this:
```tsx
    <div
      id=''
      className=''
      title=''
    >Child
    </div>
```

Instead of this:
```tsx
    <div
      id=''
      className=''
      title=''
    >
      Child
    </div>
```

### Reproduction

https://stackblitz.com/edit/github-wkwyuhjr?file=src%2Findex.tsx

### Contributes

- [x] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests

**Comment by lucharlesc** [comment]
I think you might need to use [jsx-one-expression-per-line](https://eslint.style/rules/jsx-one-expression-per-line) to force `Child` to be on a new line.

**Comment by 9romise** [comment]
The `jsx-max-props-per-line` rule specifically focuses on formatting **props within `JSXOpeningElements`**. The line breaks after the opening element should not be handled by this rule.

I'll close this since it's outside the rule's responsibilities, but please feel free to reopen if you have additional insights!

**Comment by flippidippi** [comment]
> The `jsx-max-props-per-line` rule specifically focuses on formatting **props within `JSXOpeningElements`**. The line breaks after the opening element should not be handled by this rule.
> 
> I'll close this since it's outside the rule's responsibilities, but please feel free to reopen if you have additional insights!

Yeah I'm not sure which rule's responsibility it should be or a new rule.

The problem with [jsx-one-expression-per-line](https://eslint.style/rules/jsx-one-expression-per-line) is that I don't like forcing one expression per line to 1, but I think there is a case for saying "if the containing component is multiline, then the children need to be on a new line", or something like that.

EDIT

Actually I didn't realize what the `jsx-one-expression-per-line` `"allow": "single-child"` option did till just now. That miiight be what I need.

**Comment by flippidippi** [comment]
Ok nvm lol. That `@stylistic/jsx-one-expression-per-line` `"allow": "single-child"` is not what we want, because it touches things that we are ok with being on a single line. 

✅ This is ok:
```tsx
<div>
  Up to {percent}% Off
</div>
```

❌ We don't want to enforce this:
```tsx
<div>
  Up to
  {' '}
  {percent}
  % Off
</div>
```

---

So it's really about making sure that when the closing tag is not on the same line as the opening tag, the children should be on a new line (not on the same line as the starting tag). I don't see a rule that could help me accomplish that without other unwanted side-effects.

❌ For example we wouldn't want this:
```tsx
<div className='fs-6 col-12 text-danger fw-bold'>Up to {percent}% Off
</div>
```

✅ We want this:
```tsx
<div className='fs-6 col-12 text-danger fw-bold'>
  Up to {percent}% Off
</div>
```

✅ And this is ok:
```tsx
<div className='fs-6 col-12 text-danger fw-bold'>Up to {percent}% Off</div>
```

Multiline example:

❌ We don't want this:
```tsx
<div
  className='fs-6 col-12 text-danger fw-bold'
>Up to {percent}% Off
</div>
```

✅ We want this:
```tsx
<div
  className='fs-6 col-12 text-danger fw-bold'
>
  Up to {percent}% Off
</div>
```

**Comment by 9romise** [comment]
In fact, I am not a `jsx` user myself.

I'm not sure if the close `>` of the open tag and its child are on the same line means that there is more than one expression on this line:
```ts
<div
  className='fs-6 col-12 text-danger fw-bold'
>Up to {percent}% Off
// prev line
</div>
```

So maybe that's a bug in `jsx-one-expression-per-line`?

**Comment by flippidippi** [comment]
I don't think so? Cause that's still multiple children so it will still do what my first example does, which we don't want.

We don't care about # of expressions per line, we only care that children of the multiline component are on a new line.

**Comment by flippidippi** [comment]
Should I submit a new ticket for this? Seems like we either need a new rule? Or updating an existing rule to have this option?

**Comment by flippidippi** [comment]
I think there's still an issue where it's not quite what we need. For example these example should be be ok.

```tsx
    <>
      <div>
        Up to {percent}% Off
      </div>

      <div>
        Hello World
        Up to {percent}% Off
      </div>
    </>
```

But the second one is showing a lint error and getting autofixed to something we don't want.

```tsx
    <>
      <div>
        Up to {percent}% Off
      </div>

      <div>
        Hello World
        Up to
        {' '}
        {percent}
        % Off
      </div>
    </>
```
